### DISCUSSION TITLE
Pods deployed in OKD4.8 are going to "ImagePullBackOff" state with "unauthorized: authentication required" error.

### DISCUSSION THREAD

**Discussion by reddybhavaniprasad** [okd-project/okd#1307] [General]
**Describe the bug**
I have successfully installed OKD4.8 and able to deploy applications. When I try to deploy a new application many days after the installation the pods are going to "ImagePullBackOff" state with "unauthorized: authentication required" error.

**Version**
Client Version: 4.8.0-0.okd-2021-11-14-052418
Server Version: 4.8.0-0.okd-2021-11-14-052418
Kubernetes Version: v1.21.2-1555+9e8f924492b7d7-dirty
I have installed OKD4.8 on VMWare machine using the UPI method.

**How reproducible**
Install OKD4.8 and deploy a few applications and leave the setup for some days and deploy a new application then the pod is going into "ImagePullBackOff" state with "unauthorized: authentication required" error.

**Log bundle**
`$ sudo podman pull image-registry.openshift-image-registry.svc:5000/frontend/nginx:v1.0
WARN[0000] Failed to decode the keys ["storage.options.override_kernel_check"] from "/etc/containers/storage.conf".
Trying to pull image-registry.openshift-image-registry.svc:5000/frontend/nginx:v1.0...
Error: initializing source docker://image-registry.openshift-image-registry.svc:5000/frontend/nginx:v1.0: unable to retrieve auth token: invalid username/password: unauthorized: authentication required`

I think this might be happening due to the oc login session being expired which the podman fails to authenticate the default internal registry with the old token. So, kindly share me any process to avoid this kind of behaviour

**Comment by vrutkovs** [comment]
See https://docs.okd.io/latest/registry/securing-exposing-registry.html

**Reply by reddybhavaniprasad** [reply]
@vrutkovs I am able to pull the image from the worker nodes after I login to default internal registry from worker nodes using the podman login command. But the authentication error occurs only when the  oc session is timed out after a period of time because the podman is still using the old key (oc whoami -t) to pull the image from worker nodes, which is working after a fresh login to internal registry.
My requirement is to make the worker node pull the image without the need of authenticating the internal registry again from the worker nodes.

**Reply by vrutkovs** [reply]
Wait, why are you using `podman pull` on worker node?

**Reply by reddybhavaniprasad** [reply]
I have used it just to check if the image is being properly pulled in the worker nodes but I am facing the same Authentication error.

**Reply by vrutkovs** [reply]
When pod is being created it would use serviceaccount's pull secret to create containers. Make sure serviceaccount has correct RBAC to access images - see https://docs.okd.io/latest/openshift_images/managing_images/using-image-pull-secrets.html

**Reply by reddybhavaniprasad** [reply]
I have applied the image puller policy for pulling the images across the projects and the pods are also able to pull the images for some days and they are running fine now as well. 
When I try to update the image in the app deployment, the pod is throwing unauthorized error while pulling the image. It is getting resolved only after login to the internal registry from the worker node and pull the image manually.
